### PR TITLE
Epmcuxda 6627 Models typing and validation

### PR DESCRIPTION
**Pull request by Shutya**
### What does this PR do?

Fixed some types in EURING entities and added validation with class-validation library

### How should I test this?

Navigate throw code and try to add new invalid ring.

### Jira ticket related to this PR

- [EPMCUXDA-6627](https://jira.epam.com/jira/browse/EPMCUXDA-6627)

#### Type of change

- [x] Adds new features.
- [x] Changes existing functionality.
- [ ] Deprecates features.
- [ ] Removes features.
- [ ] Fixes any bug.
- [ ] Other: security, build process, infrastructure, tests, docs, etc.

### CODE REVIEW

File: src/controllers/abstract-controller.ts [hunk 1]
```diff
@@ -1,9 +1,21 @@
 import config from 'config';
 import { Router, NextFunction, Request, Response } from 'express';
 import { Repository } from 'typeorm';
+import { validate } from 'class-validator';
 
 const UUID_LENGTH = config.get('UUID_LENGTH');
 
+class ValidationError {
```
Comment (mishelen) [inline src/controllers/abstract-controller.ts]: Or simply reuse CustomError from utils, or move it in utils

File: src/controllers/abstract-controller.ts [hunk 2]
```diff
@@ -39,5 +51,21 @@ export default abstract class AbstractController {
     }
   }
 
+  // Argument 'data' it is a new data, and argument existedData is optional and needed for refreshing existing data in db
+  protected async validate(data: any, existedData: any = {}) {
+    const createdModel = await this.entity.create(Object.assign(existedData, data));
+    const errors = await validate(createdModel);
+    if (errors.length) {
+      const parsedErrors = errors.reduce(
+        (acc, error) => ({
```
Comment (mishelen) [inline src/controllers/abstract-controller.ts]: please, add typings to reduce callback params

File: src/controllers/abstract-controller.ts [hunk 3]
```diff
@@ -39,5 +51,21 @@ export default abstract class AbstractController {
     }
   }
 
+  // Argument 'data' it is a new data, and argument existedData is optional and needed for refreshing existing data in db
+  protected async validate(data: any, existedData: any = {}) {
```
Comment (mishelen) [inline src/controllers/abstract-controller.ts]: Type returned by validate value

File: src/controllers/abstract-controller.ts (hunk 1 above)
Comment (mishelen) [reply src/controllers/abstract-controller.ts]: Also, as you had touched Abstract Controller please change native Errors to Custom in checkId

**Comment by mishelen** [comment]
Denis! Don't forget to update swagger definitions

File: src/controllers/abstract-controller.ts (hunk 1 above)
Comment (Shutya) [reply src/controllers/abstract-controller.ts]: Thank you, I've missed existence of CustomError. Done.

File: src/controllers/abstract-controller.ts (hunk 2 above)
Comment (Shutya) [reply src/controllers/abstract-controller.ts]: Done.

File: src/controllers/abstract-controller.ts (hunk 3 above)
Comment (Shutya) [reply src/controllers/abstract-controller.ts]: Done.

**Comment by Shutya** [comment]
> Denis! Don't forget to update swagger definitions

Sorry, my bad. Fixed.

**Pull request merged**